SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReplicationGraphLoadBalancingHandler.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialReplicationGraphLoadBalancingHandler.h"
+#include "EngineClasses/SpatialReplicationGraph.h"
+
+FSpatialReplicationGraphLoadBalancingContext::FSpatialReplicationGraphLoadBalancingContext(USpatialNetDriver* InNetDriver, USpatialReplicationGraph* InReplicationGraph, FPrioritizedRepList& InRepList)
+	: NetDriver(InNetDriver)
+	, ReplicationGraph(InReplicationGraph)
+	, ActorsToReplicate(InRepList)
+{
+
+}
+
+
+FSpatialReplicationGraphLoadBalancingContext::FRepListArrayAdaptor FSpatialReplicationGraphLoadBalancingContext::GetActorsBeingReplicated()
+{
+	return FRepListArrayAdaptor(ActorsToReplicate);
+}
+
+void FSpatialReplicationGraphLoadBalancingContext::RemoveAdditionalActor(AActor* Actor)
+{
+
+}
+
+void FSpatialReplicationGraphLoadBalancingContext::AddActorToReplicate(AActor* Actor)
+{
+	ReplicationGraph->ForceNetUpdate(Actor);
+}
+
+FActorRepListRefView FSpatialReplicationGraphLoadBalancingContext::GetDependentActors(AActor* Actor)
+{
+	static FActorRepListRefView s_EmptyList = []
+	{
+		FActorRepListRefView List;
+		List.Reset(0);
+		return List;
+	}();
+
+	if (FGlobalActorReplicationInfo* GlobalActorInfo = ReplicationGraph->GetGlobalActorReplicationInfoMap().Find(Actor))
+	{
+		const FActorRepListRefView& DependentActorList = GlobalActorInfo->GetDependentActorList();
+		if (DependentActorList.IsValid())
+		{
+			return DependentActorList;
+		}
+	}
+	return s_EmptyList;
+}
+
+bool FSpatialReplicationGraphLoadBalancingContext::IsActorReadyForMigration(AActor* Actor)
```

Comment:
FSpatialNetDriverLoadBalancingContext version has additional checks. Shouldn't they be the same?

Reply:
Can we move the shared logic to somewhere shared, and explicitly call out in code the differences that exist. I'm concerned about minimizing the duplication to minimize the likelihood of bugs that affect only one replication approach.

Reply:
Added comments. 
I think the only shared part is the auth + readiness check.